src/Microsoft.ApplicationInsights/TelemetryClientExtensions.cs

```diff
@@ -159,14 +155,15 @@ public static IOperationHolder<T> StartOperation<T>(this TelemetryClient telemet
                     operationActivity.SetOperationName(operationName);
                 }
 
-                if (parentActivity == null)
+                if (idsAssignedByUser)
                 {
                     // telemetryContext.Id is always set: if it was null, it is set to newly generated W3C compatible TraceID
                     if (Activity.DefaultIdFormat == ActivityIdFormat.W3C)
                     {
                         // There is no need of checking if TelemetryContext.ID is W3C Compatible. It is always set to 
                         // W3C compatible id. Even user supplied non-compatible ID is ignored.
-                        operationActivity.SetParentId(ActivityTraceId.CreateFromString(telemetryContext.Id.AsSpan()), default(ActivitySpanId), ActivityTraceFlags.None);
+                        operationActivity.SetParentId(ActivityTraceId.CreateFromString(telemetryContext.Id.AsSpan()),
```

Comment:
StartOperation<T>(this TelemetryClient telemetryClient, T operationTelemetry) is public, so we need to validate if operationTelemetry.Context.Operation.ID is W3c compatible before taking it without which exception is thrown by activity. Need similar check as done for other overload of startoperation.